samcli/lib/telemetry/event.py

```diff
@@ -84,6 +85,7 @@ class EventTracker:
     """Class to track and recreate Events as they occur."""
 
     _events: List[Event] = []
+    _event_lock = threading.Lock()
```

Comment:
I did consider whether the Lock would need to be tested, but given the simplicity and straightforwardness of the `threading` module using the context manager, I didn't think it would be necessary.

Reply:
👍 sounds good. I think existing tests should just work the same

Reply:
We shouldn't be testing if the locking will work, but we can test like certain features will acquire the lock by mocking them. I am approving this to unblock but these tests could be something that you can add it later.